### ISSUE TITLE
Rule Change: quote-props

### ISSUE THREAD

**Issue by betandbuzz** [eslint/eslint#15272]
### What rule do you want to change?

quote-props

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
"myMethod"() {
    ...
}
```

### What does the rule currently do for this code?

No error.

### What will the rule do after it's changed?

Unnecessarily quoted property 'myMethod' found.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @betandbuzz, thanks for the issue!

Since this isn't a bug, this change would require a new option, but [stylistic rules are frozen](https://github.com/eslint/eslint#stylistic-rule-updates) so we won't be adding any new options to the `quote-props` rule. You can always copy the existing rule into a custom rule and make it behave the way you want.
